test/attributeInstanceIn.lean

```diff
@@ -53,24 +51,44 @@ attribute [instance] Int.add in
 instance : Inhabited Int where
   default := 0
 
+set_option linter.attributeInstanceIn false in
 #guard_msgs in
--- non-`instance` attributes are allowed with `in`
+/--
+warning: Despite the `in`, the attribute 'simp' is added globally to 'Int.add'
+please remove the `in` or make this a `local instance` instead
+note: this linter can be disabled with `set_option linter.attributeInstanceIn false`
+-/
 #guard_msgs in
+set_option linter.attributeInstanceIn true in
 attribute [simp] Int.add in
 instance : Inhabited Int where
   default := 0
 
+namespace X
+
 -- Here's another example, with nested attributes.
-def foo := True
+/-- warning: declaration uses 'sorry' -/
+#guard_msgs in
+theorem foo (x y : Nat) : x = y := sorry
 
+set_option linter.attributeInstanceIn false in
+#guard_msgs in
+/--
+warning: Despite the `in`, the attribute 'simp' is added globally to 'foo'
+please remove the `in` or make this a `local instance` instead
+note: this linter can be disabled with `set_option linter.attributeInstanceIn false`
+---
+warning: Despite the `in`, the attribute 'ext' is added globally to 'foo'
+please remove the `in` or make this a `local instance` instead
+note: this linter can be disabled with `set_option linter.attributeInstanceIn false`
+-/
 #guard_msgs in
 -- TODO: where's the docBlame linter defined? what do I need to import?
```

Comment:
In `Batteries.Tactic.Lint.Misc`

Reply:
I agree that adding it would be a good test.

Reply:
Unfortunately, I am not sure what should be done here with the `docBlame` linter... :upside_down_face:

Reply:
I don't understand why `[nolint docBlame]` is something we care about.